sklearn/utils/_plotting.py

```diff
@@ -100,3 +100,51 @@ def _interval_max_min_ratio(data):
     """
     diff = np.diff(np.sort(data))
     return diff.max() / diff.min()
+
+
+def _validate_style_kwargs(default_style_kwargs, user_style_kwargs):
+    """Create valid style kwargs by avoiding Matplotlib alias errors.
+
+    Matplotlib raises an error when, for example, 'color' and 'c', or 'linestyle' and
+    'ls', are specified together. To avoid this, we automatically keep only the one
+    specified by the user and raise an error if the user specifies both.
+
+    Parameters
+    ----------
+    default_style_kwargs : dict
+        The Matplotlib style kwargs used by default in the scikit-learn display.
+    user_style_kwargs : dict
+        The user-defined Matplotlib style kwargs.
+
+    Returns
+    -------
+    valid_style_kwargs : dict
+        The validated style kwargs taking into account both default and user-defined
+        Matplotlib style kwargs.
+    """
+
+    invalid_to_valid_kw = {
```

Comment:
May I know where this mapping came from? I searched in the matplotlib source code where they used their `define_aliases` decorator, and there seems to be many more that we missed here (though maybe not all of them are related here).

https://github.com/search?q=repo%3Amatplotlib%2Fmatplotlib+define_aliases&type=code

Reply:
Yes I made it myself, more or less arbitrarily. But since matplotlib has a clearly defined list of aliases, maybe let's just put the entire list instead?

Reply:
Yes indeed. We could even import it because we expect to have matplotlib installed.

Reply:
I did a quick check, and I'm not sure how to import it in practice?

Reply:
Yep, I could not find a non-private way of getting them. So I would define the list such that in account for the keys/values of the matplotlib artist that we are using.

FYI, our function is close to this function (https://github.com/matplotlib/matplotlib/blob/v3.9.2/lib/matplotlib/cbook.py#L1795-L1848) when it comes to matplotlib itself. No need to try using it thought.

Reply:
So I've added all the aliases, except for those used for 3D plotting

Reply:
LGTM. Thanks @JosephBARBIERDARNAL